src/balancereader/src/main/java/anthos/samples/bankofanthos/balancereader/LedgerReader.java

```diff
@@ -73,7 +74,8 @@ public void startWithCallback(LedgerReaderCallback callback)
         this.callback = callback;
         // get the latest transaction id in ledger
         try {
-            this.latestId = dbRepo.latestId();
+            Long dbId = dbRepo.latestId();
+            this.latestId = (dbId != null ? dbId : this.latestId);
```

Comment:
I personally find these a bit hard to read. 

I'd say the could be done cleaner with a null check, since the else case is a no-op:
```
if (dbId != null){
   this.latestId = dbId
}
```

It's also a bit hard to read with 3 similarly named variables interacting with each other. It might be better if you could remove the temporary `dbId` variable (but this would require some changes to handle/remove nulls later)
```
this.latestId = dbRepo.latestId();
```

Or maybe just some extra comments would help

But these are just style issues, so feel free to ignore

Reply:
Roger that.  I agree, it is hard to read and understand.  So I made some readability changes that I think makes the code easier to understand.